### DISCUSSION TITLE
Best way to run large number of power flows

### DISCUSSION THREAD

**Discussion by vokabel123** [e2nIEE/pandapower#1610] [Q&A]
Hi, I'm working on sensitivity analysis for power systems. For this, I need to run many power flows with different values of the inputs, for example, 1000 different pairs of values for P and Q at every load. I was wondering what would be the best way to do this with pandapower, as I understand I cannot have some kind of network dictionary with all the different variations. So far, my two options are using a time series approach and just having a for loop updating the P and Q value and then running the power flow at every iteration.

Any help would be appreciated

**Comment by owen1gibson** [comment]
I needed to do something similar and ran the load flows in parallel. The python concurrent.futures module helped here (https://docs.python.org/3/library/concurrent.futures.html). For each load flow I had a separate thread in which I updated the loads then ran a load flow. I found that for over 20 load flows I had a massive reduction in total time taken to run all load flows, about 80% time reduction when I ran 200 load flows. Also watch out for how you are saving the output data, I found that to be a time sink (https://pandapower.readthedocs.io/en/v2.10.1/file_io.html)

**Comment by pajjaecat** [comment]
In the same logic as the previous answer, you could also use [``ipyparallel``](https://ipyparallel.readthedocs.io/en/latest/). Let's say you create N total engines; since you have 1000 pairs of values,  each engine will run 1000/N loops which will definitely be faster than a unique loop.

**Comment by AnkurArohi** [comment]
It also depends on what kind of sensitivity analysis you are trying to achieve, If for you the P sensitivities are enough I would suggest to use make_PTDF function or else, another way would be to use the Jacobian matrix so you do not run multiple power flows but actually a single one
